alfresco-docker-miscellaneous-transformers/src/main/java/org/alfresco/transformer/transformers/AppleIWorksContentTransformer.java

```diff
@@ -0,0 +1,176 @@
+ * parsing HTML content.
+ *
+ * <p>
+ * This code is based on a class of the same name originally implemented in alfresco-repository.
+ * </p>
+ *
+ * <p>
+ * Since HTML Parser was updated from v1.6 to v2.1, META tags
+ * defining an encoding for the content via http-equiv=Content-Type
+ * will ONLY be respected if the encoding of the content item
+ * itself is set to ISO-8859-1.
+ * </p>
+ *
+ * <p>
+ * Tika Note - could be converted to use the Tika HTML parser,
+ *  but we'd potentially need a custom text handler to replicate
+ *  the current settings around links and non-breaking spaces.
+ * </p>
+ *
+ * @see <a href="http://htmlparser.sourceforge.net/">http://htmlparser.sourceforge.net</a>
+ * @see org.htmlparser.beans.StringBean
+ * @see <a href="http://sourceforge.net/tracker/?func=detail&aid=1644504&group_id=24399&atid=381401">HTML Parser</a>
+ *
+ * @author Derek Hulley
+ * @author eknizat
+ */
+@Component
+public class AppleIWorksContentTransformer extends AbstractJavaTransformer
+{
+    private static final Logger logger = LoggerFactory.getLogger(AppleIWorksContentTransformer.class);
+
+    public static final String SOURCE_MIMETYPE = "sourceMimetype";
+    public static final String SOURCE_EXTENSION = "sourceExtension";
+    public static final String TARGET_MIMETYPE = "targetMimetype";
+    public static final String TARGET_EXTENSION = "targetExtension";
+
+    public static final List<String> REQUIRED_OPTIONS = Arrays.asList(
+            SOURCE_MIMETYPE,
+            SOURCE_EXTENSION,
+            TARGET_MIMETYPE,
+            TARGET_EXTENSION);
+
+
+    // Apple's zip entry names for previews in iWorks have changed over time.
+    private static final List<String> PDF_PATHS = Arrays.asList(
+            "QuickLook/Preview.pdf");  // iWorks 2008/9
+    private static final List<String> JPG_PATHS = Arrays.asList(
+            "QuickLook/Thumbnail.jpg", // iWorks 2008/9
+            "preview.jpg");            // iWorks 2013/14 (720 x 552) We use the best quality image. Others are:
+    //                (225 x 173) preview-web.jpg
+    //                 (53 x  41) preview-micro.jpg
+
+    private static final List<String> IWORKS_MIMETYPES = Arrays.asList(Mimetype.MIMETYPE_IWORK_KEYNOTE,
+            Mimetype.MIMETYPE_IWORK_NUMBERS,
+            Mimetype.MIMETYPE_IWORK_PAGES);
+    private static final List<String> TARGET_MIMETYPES = Arrays.asList(Mimetype.MIMETYPE_IMAGE_JPEG
+// Commented out rather than removed, in case we can get SHARE to fall back to using JPEG when a PDF is not available
+//                                                                    ,MimetypeMap.MIMETYPE_PDF
+    );
+
+    @Autowired
+    public AppleIWorksContentTransformer(SelectingTransformer miscTransformer)
+    {
+        super(miscTransformer);
+    }
+
+    @Override
+    public boolean isTransformable(String sourceMimetype, String targetMimetype, Map<String, String> parameters)
+    {

```

Comment:
If I remember correctly the new transformers don't do isTransformable in Java code. It should be checked via configuration. This appears to be code copied from the legacy transformers. As a result the static finals above can also be removed.